segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -438,20 +439,24 @@ public void createSegment(CreateSegment createStreamSegment) {
         Timer timer = new Timer();
         final String operation = "createSegment";
 
+        Preconditions.checkArgument(createStreamSegment.getRolloverSizeBytes() >= 0,
+                "Segment rollover size must not be negative; actual %s.", createStreamSegment.getRolloverSizeBytes());
+
         Collection<AttributeUpdate> attributes = Arrays.asList(
                 new AttributeUpdate(SCALE_POLICY_TYPE, AttributeUpdateType.Replace, ((Byte) createStreamSegment.getScaleType()).longValue()),
                 new AttributeUpdate(SCALE_POLICY_RATE, AttributeUpdateType.Replace, ((Integer) createStreamSegment.getTargetRate()).longValue()),
+                new AttributeUpdate(ROLLOVER_SIZE, AttributeUpdateType.Replace, createStreamSegment.getRolloverSizeBytes()),
```

Comment:
If `0`, leave it as "default" - that is, do not add it.

Reply:
I'm not sure about this one. `Arrays.asList` is unmodifiable so if we keep it as is when 0 then the code would be ugly. Do you think we can make `attributes` modifiable like using `ArrayList`?